### DISCUSSION TITLE
Is there a way to use multiple Font providers?

### DISCUSSION THREAD

**Discussion by erkage** [unocss/unocss#3568] [Q&A] [answered]
Hey,

So, I'd like to use more presetWebFonts provider.
I've tried it with the following code:
```
presetWebFonts(
   {
      provider: 'fontshare',
      fonts: {
         satoshi: 'Satoshi:400,500,700',
      },
   },
   {
      provider: 'google',
      fonts: {
         signika: 'Signika:300,400,500,700',
      },
   }
)
```
but it seems it only picks up the first one.

**Comment by zyyv** [comment]
I'd love to make this happen.

**Reply by erkage** [reply]
That would be awesome 🙂.
I'm surprised that this hasn't came up yet.
Also checking my attempt with a fresh head I don't know why I tried that and not using an array of objects 🙂.

**Reply by zyyv** [reply, marked as answer]
You can custom the webfont `meta` in `fonts` option, override the default font provider.
https://github.com/unocss/unocss/blob/cc5fabbd106770d50c682dc37a4437ae3279cf5e/packages/preset-web-fonts/src/types.ts#L7-L11

e.g. [Playground](https://unocss.dev/play/?html=DwEwlgbgBAxgNgQwM5ILwCIDuAnBAHPAU23QD4AoKKUSWRFDAMwHsA7AFwFokF3mkAFmDIBVVswDCAZSnAA9OAgUqNaPGRp0LDtzABzVmADWCUeOmyFkCvMUUgA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAqrhJajfYwOoYBGAMWIw6lAL5wsUCCDgByAK6sAxnTpykSDAA9IsdNgCGCgDbxMOfERLBSACmSo6AC2gxlCkQC5EVVAHcoQzAwDCgfOX8AWjplWgxcOGcYuIwEyRMdOAArBToYYCwATyjlBJgwuGAKkDpS8sqsTO1SiBM5djgxTs4GbzgAbT9qWj6WCDsASk6OUZ5%2BIVwRB2HZiAA3YExw%2BVIICFJMjtXJYTofR1Qrp0MYCBdgC5Pr3EMQDAiAZVv752Bj66AuD%2BDC2Zz9AZyAAsAAYYR15ABWOEIuQAdhRAF0ZkCRhstmEIlgzs5DLQAbjus84HRbLhgABrQxfOmM5kAZjh5FhMPIyN5GPhONQVOuYmmVGxSHFSCAA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmsBAA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwBpxp5k0BhAVzjCVQoGVHNs8iTZEVUAJMVDla4CAXyA)

I'll send a PR in my free time to update `WebFontsOptions` to `Arrayable<WebFontsOptions>`

**Reply by erkage** [reply]
Ahh, great, thank you (both the solution and the work on the Array way)!

**Reply by xarthurx** [reply]
> You can custom the webfont `meta` in `fonts` option, override the default font provider.
> 
> https://github.com/unocss/unocss/blob/cc5fabbd106770d50c682dc37a4437ae3279cf5e/packages/preset-web-fonts/src/types.ts#L7-L11
> 
> e.g. [Playground](https://unocss.dev/play/?html=DwEwlgbgBAxgNgQwM5ILwCIDuAnBAHPAU23QD4AoKKUSWRFDAMwHsA7AFwFokF3mkAFmDIBVVswDCAZSnAA9OAgUqNaPGRp0LDtzABzVmADWCUeOmyFkCvMUUgA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAqrhJajfYwOoYBGAMWIw6lAL5wsUCCDgByAK6sAxnTpykSDAA9IsdNgCGCgDbxMOfERLBSACmSo6AC2gxlCkQC5EVVAHcoQzAwDCgfOX8AWjplWgxcOGcYuIwEyRMdOAArBToYYCwATyjlBJgwuGAKkDpS8sqsTO1SiBM5djgxTs4GbzgAbT9qWj6WCDsASk6OUZ5%2BIVwRB2HZiAA3YExw%2BVIICFJMjtXJYTofR1Qrp0MYCBdgC5Pr3EMQDAiAZVv752Bj66AuD%2BDC2Zz9AZyAAsAAYYR15ABWOEIuQAdhRAF0ZkCRhstmEIlgzs5DLQAbjus84HRbLhgABrQxfOmM5kAZjh5FhMPIyN5GPhONQVOuYmmVGxSHFSCAA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmsBAA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwBpxp5k0BhAVzjCVQoGVHNs8iTZEVUAJMVDla4CAXyA)
> 
> I'll send a PR in my free time to update `WebFontsOptions` to `Arrayable<WebFontsOptions>`

Has this feature been implemented?
For multi-language website, this feature is very important.

**Reply by zyyv** [reply]
This feature has always existed, please refer to the documentation or test cases.
